### PR TITLE
[Prompt] PromptModelForSequenceClassification and unit tests

### PR DESCRIPTION
**Pull request by sijunhe**
### PR types
Function optimization

### PR changes
APIs

### Description
- Modify API of `PromptModelForSequenceClassification` to behave similarly to `AutoModelForSequenceClassification`
- Add unit tests

### CODE REVIEW

File: paddlenlp/prompt/prompt_model.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+        if not return_dict:
+            output = (logits, hidden_states)
+            return ((loss,) + output) if loss is not None else output
+
+        return SequenceClassifierOutput(
+            loss=loss,
+            logits=logits,
+            hidden_states=hidden_states,
+        )
```
Comment (sijunhe) [inline paddlenlp/prompt/prompt_model.py]: same logic as `AutoModelForSequenceClassification`

File: paddlenlp/prompt/prompt_trainer.py [hunk 2]
```diff
@@ -43,10 +40,10 @@ class PromptTrainer(Trainer):
 
     def __init__(
         self,
-        model: Union[nn.Layer],
+        model: nn.Layer,
         tokenizer: PretrainedTokenizer,
-        criterion: Union[nn.Layer],
-        args: PromptTuningArguments = None,
+        criterion: Optional[nn.Layer] = None,
```
Comment (sijunhe) [inline paddlenlp/prompt/prompt_trainer.py:45]: similar behavior as `Trainer`

File: tests/prompt/test_prompt_model.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    PromptDataCollatorWithPadding,
+    PromptModelForSequenceClassification,
+    SoftVerbalizer,
+)
+from paddlenlp.transformers import AutoModelForMaskedLM, AutoTokenizer
+
+
+class TestPromptModel(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
```
Comment (LemonNoel) [inline tests/prompt/test_prompt_model.py:32]: 建议新增一个单测，模型输入 `model: AutoModelForSequenceClassification`, `verbalizer: None`，保证类似 [EFL](https://github.com/PaddlePaddle/PaddleNLP/blob/develop/examples/few_shot/efl/run_train.py) 的算法能正常执行。

Comment (sijunhe) [reply tests/prompt/test_prompt_model.py:32]: 已加入

File: tests/prompt/test_prompt_model.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        cls.label_words = {0: "0", 1: "1", 2: "2"}
+        cls.verbalizer = SoftVerbalizer(cls.label_words, cls.tokenizer, cls.model)
+        cls.data_collator = PromptDataCollatorWithPadding(cls.tokenizer, padding=True, return_tensors="pd")
+        cls.prompt_model = PromptModelForSequenceClassification(cls.model, cls.template, cls.verbalizer)
+
+    def test_sequence_classification_no_labels(self):
+        examples = [{"text": "百度飞桨深度学习框架"}, {"text": "这是一个测试"}]
+        encoded_examples = [self.template(i) for i in examples]
+        logits, hidden_states = self.prompt_model(**self.data_collator(encoded_examples))
+        self.assertTrue(logits.shape[0], len(examples))
```
Comment (LemonNoel) [inline tests/prompt/test_prompt_model.py]: It seems that `assertTrue` only works for boolean type.

File: tests/prompt/test_prompt_model.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (LemonNoel) [inline tests/prompt/test_prompt_model.py:1]: You can add one line in Makefile to run the unittest.

```
unit-test:
	# only enable bert-test: there are many failed tests
	PYTHONPATH=$(shell pwd) pytest tests/transformers/bert
	PYTHONPATH=$(shell pwd) pytest tests/prompt
```

**Review by LemonNoel — CHANGES_REQUESTED** [review]

File: paddlenlp/prompt/prompt_model.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+            model_inputs.pop("masked_positions")
+        hidden_states = self.plm(**model_inputs)
+        if self.verbalizer is not None:
+            logits = self.verbalizer.process_outputs(hidden_states, input_dict["masked_positions"])
+        else:
+            logits = hidden_states
+
+        loss = None
+        if labels is not None:
+            if self.verbalizer is None:
```
Comment (LemonNoel) [inline paddlenlp/prompt/prompt_model.py]: Directly raising Error when self.verbalizer is None will make `PromptModelForSequenceClassification` not applicable for EFL in `PromptTrainer`, since labels are always passed to `PromptModelForSequenceClassification`.

Maybe the condition can be modified as something like

```
if self.verbalizer is None and not isinstance(self.plm, xxForSeuquenceClassification)
```

**Review by LemonNoel — CHANGES_REQUESTED** [review]

File: paddlenlp/prompt/prompt_model.py (hunk 6 above)
Comment (LemonNoel) [reply paddlenlp/prompt/prompt_model.py]: Or type checking could be added in initialization and it makes sure that self.plm is either `xxForSequenceClassification` or ``xxForMaskedLM``. Then even if self.verbalizer is None, the loss can also be calculated here with CE Loss.

File: paddlenlp/prompt/prompt_trainer.py [hunk 7]
```diff
@@ -64,6 +61,9 @@ def __init__(
         if data_collator is None:
             data_collator = PromptDataCollatorWithPadding(tokenizer, padding=True, return_tensors="pd")
 
+        if criterion is None and (args.use_rgl or args.use_rdrop):
+            raise Exception("'To use 'use_rgl', use_rdrop', 'criterion' must be specified")
```
Comment (LemonNoel) [inline paddlenlp/prompt/prompt_trainer.py]: It seems that the first single quotation mark is in the wrong place.

File: tests/prompt/test_prompt_model.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    SoftVerbalizer,
+)
+from paddlenlp.transformers import (
+    AutoModelForMaskedLM,
+    AutoModelForSequenceClassification,
+    AutoTokenizer,
+)
+
+
+class TestPromptModel(unittest.TestCase):
```
Comment (LemonNoel) [inline tests/prompt/test_prompt_model.py]: According to @wj-Mcat , the class name of TestCase follows the rule: {name}Test, you can refer to: [transformers unittest](https://github.com/PaddlePaddle/PaddleNLP/blob/develop/tests/transformers/roberta/test_modeling.py#L399).

**Review by LemonNoel — CHANGES_REQUESTED** [review]

**Review by LemonNoel — APPROVED** [review]
LGTM
